File: Exercism/php/bob/bob.php
<?php
class Bob
{
    public function respondTo(string $_s): string
    {
        // Get rid of whitespaces
        $_s = preg_replace('/\s+/', '', $_s);

        // check if all chars are uppercase (shouting)
        $shouting = true;
        for ($i = 0; $i < strlen($_s); $i++) {
            if (ctype_lower($_s[$i])) {
                $shouting = false;
                break;
            }
        }

        // check if string contains numbers (and punctual chars) only
        $numOnly = true;
        for ($i = 0; $i < strlen($_s); $i++) {
            if (!ctype_digit($_s[$i]) && !ctype_punct($_s[$i])) {
                $numOnly = false;
                break;
            }
        }

        if ($shouting && mb_substr($_s, -1) == '?' && !$numOnly) {
            return 'Calm down, I know what I\'m doing!';
        } else if ($shouting && !$numOnly && strlen($_s) > 0) {
            return 'Whoa, chill out!';
        } else if (mb_substr($_s, -1) == '?') {
            return 'Sure.';
        } else if (strlen($_s) == 0) {
            return 'Fine. Be that way!';
        }

        return 'Whatever.';
    }
}

File: Exercism/cpp/bob/bob.cpp
#include "bob.h"

string bob::hey(string s) {
  // get rid of white spaces
  s.erase(remove(s.begin(), s.end(), ' '), s.end());

  // get rid of white printing characters
  s.erase(s.begin(), find_if(s.begin(), s.end(), [](unsigned char ch) {
    return !isspace(ch);
  }));

  // check if all chars are uppercase (shouting)
  bool shouting = true;
  for(unsigned int i = 0; i < s.length(); i++) {
    if(!isupper(s[i]) && !ispunct(s[i]) && !isdigit(s[i])) {
      shouting = false;
      break;
    }
  }

  // check if string contains numbers (and punctual chars) only
  bool numOnly = true;
  for(unsigned int i = 0; i < s.length(); i++) {
    if(!isdigit(s[i]) && !ispunct(s[i])) {
      numOnly = false;
      break;
    }
  }

  if(shouting && !numOnly && s.back() == '?')
    return "Calm down, I know what I'm doing!";
  else if(shouting && s.length() > 0 && !numOnly)
    return "Whoa, chill out!";
  else if(s.back() == '?')
    return "Sure.";
  else if(s.length() == 0)
    return "Fine. Be that way!";
  
  return "Whatever.";
}
